lib/lexer/src/lexer.rs

```diff
@@ -406,24 +406,29 @@ impl Lexer {
     fn consume_digits(&mut self, digit_characters: String, token: &mut String) {
         loop {
             // Try consume '_' digit.
-            let character = self.peek_char(None);
-            if character.is_some() && character.unwrap() == '_' {
+            let mut character = self.peek_char(None);
+            while character.is_some() && character.unwrap() == '_' {
                 // Consume '_'.
                 self.eat_char();
+                character = self.peek_char(None);
 
-                // If '_' is consumed, a digit must follow.
-                let character = self.peek_char(None);
-                if character.is_some() && digit_characters.find(character.unwrap()).is_some() {
-                    token.push(self.eat_char());
-                } else { // Otherwise spit out '_' and break.
-                    self.cursor -= 1;
-                    break;
-                }
-            } else if character.is_some() && digit_characters.find(character.unwrap()).is_some() {
+            }
+
+            // If '_' is consumed, a digit must follow.
+            let character = self.peek_char(None);
+            if character.is_some() && digit_characters.find(character.unwrap()).is_some() {
                 token.push(self.eat_char());
-            } else {
+            } else { // Otherwise spit out '_' and break.
```

Comment:
As a follow up, you probably need to maintain a count while consuming the underscores.
```rust

let mut character = self.peek_char(None);
let mut count = 0;
while character.is_some() && character.unwrap() == '_' {
    // Consume '_'.
    self.eat_char();
    character = self.peek_char(None);
    count += 1;
}

// If '_' is consumed, a digit must follow.
let character = self.peek_char(None);
if character.is_some() && digit_characters.find(character.unwrap()).is_some() {
    token.push(self.eat_char());
} else { // Otherwise spit out '_' and break.
    self.cursor -= count;
    break;
}
```

Reply:
yh thats what i did